src/util/SubscriptionPromise.ts

```diff
@@ -0,0 +1,63 @@
+import { ERROR_TIMEOUT } from '../errorhandling/SDKErrors'
+
+export interface Evaluator<I, O> {
+  (value: I): [boolean, O]
+}
+
+export function makeSubscriptionPromise<SubscriptionType, PromiseType>(
+  resolveOn: Array<Evaluator<SubscriptionType, PromiseType>>,
```

Comment:
Let's just have one Evaluator (which internally can combine different conditions)

Reply:
I have simplified the evaluator, it's now way easier to just chain multiple evaluators to produce a combined one. You'll like it!